File: library/src/main/java/com/snail/commons/base/BaseHolder.java
package com.snail.commons.base;

import android.support.annotation.NonNull;
import android.view.View;

/**
 * 主要用于ListView的item布局创建及数据设置
 */
public abstract class BaseHolder<T> {
	private View convertView;

	public BaseHolder() {
		convertView = createConvertView();
		convertView.setTag(this);
	}

	public View getConvertView() {
		return convertView;
	}

	/**
	 * 设置数据
	 */
	protected abstract void setData(@NonNull T data, int position);
	
	/**
	 * 创建界面
	 */
	protected abstract View createConvertView();
}

File: library/src/main/java/com/snail/commons/base/BaseHolder.kt
package com.snail.commons.base

import android.view.View

/**
 * 主要用于ListView的item布局创建及数据设置
 */
@Suppress("LeakingThis")
abstract class BaseHolder<T> {
    val convertView: View

    init {
        convertView = createConvertView()
        convertView.tag = this
    }

    /**
     * 设置数据
     */
    abstract fun setData(data: T, position: Int)

    /**
     * 创建界面
     */
    protected abstract fun createConvertView(): View
}